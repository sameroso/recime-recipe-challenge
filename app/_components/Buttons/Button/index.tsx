"use client";

import styled from "styled-components";
import { IButton } from "../types";

export const Button: IButton = styled.button`
  cursor: pointer;
`;
