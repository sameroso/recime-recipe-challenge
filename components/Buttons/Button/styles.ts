import { css, RuleSet } from "@/libs/styled-components";
import { ButtonSizes, ButtonVariants } from "../types";

type ConncatDefaultToUnion<TUnion extends string> = TUnion | "default";

const PrimaryButton = css`
  background-color: ${({ theme }) => theme.colors.primary};
  color: white;
`;

const SmallButton = css`
  padding: 10px 10px;
`;

const MediumButton = css`
  padding: 10px 20px;
`;

const LargeButton = css`
  padding: 10px 30px;
`;

export const buttonVariants: Record<
  ConncatDefaultToUnion<ButtonVariants>,
  RuleSet<object>
> = {
  primary: PrimaryButton,
  default: PrimaryButton,
};

export const buttonSizes: Record<
  ConncatDefaultToUnion<ButtonSizes>,
  RuleSet<object>
> = {
  large: LargeButton,
  medium: MediumButton,
  small: SmallButton,
  default: MediumButton,
};
