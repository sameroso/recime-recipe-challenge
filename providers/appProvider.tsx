import {
  GlobalStyle,
  recipeTheme,
  ThemeProvider,
} from "@/libs/styled-components";
import StyledComponentsRegistry from "@/libs/styled-components/registry";
import { ReactQueryProvider } from "@/libs/react-query/providers";

export function AppProvider({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <ThemeProvider theme={recipeTheme}>
      <ReactQueryProvider>
        <StyledComponentsRegistry>
          <GlobalStyle />
          {children}
        </StyledComponentsRegistry>
      </ReactQueryProvider>
    </ThemeProvider>
  );
}
