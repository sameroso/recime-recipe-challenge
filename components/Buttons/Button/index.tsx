"use client";
import { styled, css } from "@/libs/styled-components";
import { IButton } from "../types";
import { buttonSizes, buttonVariants } from "./styles";

export const Button: IButton = styled.button`
  cursor: pointer;
  border-radius: ${({ theme }) => theme.borderRadius.medium};
  ${({ $variant }) => buttonVariants[$variant || "default"]}

  ${({ $size }) => buttonSizes[$size || "default"]}
`;
