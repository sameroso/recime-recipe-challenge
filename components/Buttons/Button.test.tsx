import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import "jest-styled-components";
import { AppProvider } from "@/providers/appProvider";
import { Button } from "./Button";

describe("Page", () => {
  it("should snapshot a  medium button", () => {
    const { container } = render(
      <AppProvider>
        <Button />
      </AppProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it("should snapshot a medium button", () => {
    const { container } = render(
      <AppProvider>
        <Button $size="medium" />
      </AppProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it("should snapshot a large button", () => {
    const { container } = render(
      <AppProvider>
        <Button $size="large" />
      </AppProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it("should snapshot a small button", () => {
    const { container } = render(
      <AppProvider>
        <Button $size="small" />
      </AppProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it("should snapshot a primary button", () => {
    const { container } = render(
      <AppProvider>
        <Button $variant="primary" />
      </AppProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
