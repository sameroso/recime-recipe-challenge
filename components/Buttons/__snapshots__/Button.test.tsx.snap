// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should snapshot a  medium button 1`] = `
.c0 {
  cursor: pointer;
  border-radius: 9px;
  background-color: #4697EB;
  color: white;
  padding: 10px 20px;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`Page should snapshot a large button 1`] = `
.c0 {
  cursor: pointer;
  border-radius: 9px;
  background-color: #4697EB;
  color: white;
  padding: 10px 30px;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`Page should snapshot a medium button 1`] = `
.c0 {
  cursor: pointer;
  border-radius: 9px;
  background-color: #4697EB;
  color: white;
  padding: 10px 20px;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`Page should snapshot a primary button 1`] = `
.c0 {
  cursor: pointer;
  border-radius: 9px;
  background-color: #4697EB;
  color: white;
  padding: 10px 20px;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`Page should snapshot a small button 1`] = `
.c0 {
  cursor: pointer;
  border-radius: 9px;
  background-color: #4697EB;
  color: white;
  padding: 10px 10px;
}

<div>
  <button
    class="c0"
  />
</div>
`;
